aten/src/ATen/native/cuda/LogAddExpKernel.cu

```diff
@@ -5,30 +5,111 @@
 #include <ATen/native/TensorIterator.h>
 #include <ATen/native/BinaryOps.h>
 #include <ATen/OpMathType.h>
+#include <ATen/native/cuda/jit_utils.h>
 #include <c10/util/MathConstants.h>
+#include <ATen/native/cuda/JitLoops.cuh>
 
 // NOTE: CUDA on Windows requires that the enclosing function
 // of a __device__ lambda not have internal linkage.
 
 namespace at::native {
 
+#if AT_USE_JITERATOR()
+
+const auto logaddexp_string = jiterator_stringify(
+  // custom min and max to be used in logcumsumexp for complex arguments
+  template <typename scalar_t, bool min>
+  std::complex<scalar_t> _logcumsumexp_minmax(const std::complex<scalar_t>& x, const std::complex<scalar_t>& y) {
```

Comment:
this should be a `__device__` function probably? Also, nvrtc won't let you use tuples, but you can do old-style pointer args
`__device__ fn(const T& x, const T& y, T * min, T * max)` to avoid calling the fn twice.